docs/spec_draft.md

```diff
@@ -3080,6 +3082,81 @@ More formally, for each element `x`: `real(x) = is_complex(x) ? x.real : x`.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.recv
+
+### Semantics
+
+Receives data into a partition from a [stablehlo.send](#stablehlosend)
+instruction, in another partition, that shares the same channel handle pair
+{`channel_id`, `channel_type`}.
+
+`stablehlo.recv`, paired with the corresponding [stablehlo.send](#stablehlosend)
+instruction, represents synchronous communication. However, in order to enable
+asynchronous data transfers, it is decomposed into two HLO instructions
+(`Recv` and `RecvDone`). Similarly, the paired
+[stablehlo.send](#stablehlosend) is internally decomposed into two HLO
+instructions (`Send` and `SendDone`). [todo](https://github.com/openxla/stablehlo/issues/642)
+
+These 4 instructions, each associated with a particular channel, has the
+following semantics:
+
+  * **Recv**: Initiator of the channel by allocating the receive buffer and
+              notifying the sender that it is ready to receive data.
+  * **Send**: Waits for the Recv notification and starts the data transfer to
+              the receive buffer.
+  * **SendDone**: Waits until the Send operation is done, so that the sender
+                  buffer can be reused for other purposes.
+  * **RecvDone**: waits until the data transfer is done and the receive buffer
+                  is ready for use.
+
+The diagram shows the execution order of the instructions.
+
+<img align="center" src="spec_draft/happens-before.svg" />
+
+The `channel_type` type has the following semantics:
+
+  * `DEVICE_TO_DEVICE`: A channel for sending data between devices.
+  * `DEVICE_TO_HOST`: A channel for sending data from the device to the host.
+                      Can only be used with a `stablehlo.send` operation.
+  * `HOST_TO_DEVICE`: A channel for sending data from the host to the device.
+                      Can only be used with a `stablehlo.recv` operation.
+
+If `is_host_transfer`, an optional input with default value as `fasle`, is
+`true`, then the operation transfers data from the host.
```

Comment:
Reword suggestion:
`is_host_transfer` is an optional input that defaults to `false`, if set to `true` the operation transfers data from the host.

Reply:
reworded by removing the the implementation detail that it is an optional input.